ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CommonStorageBackend.scala

```diff
@@ -524,6 +525,32 @@ private[backend] trait CommonStorageBackend[DB_BATCH] extends StorageBackend[DB_
             )"""
     }(connection, loggingContext)
 
+    if (pruneAllDivulgedContracts) {
+      val pruneAfterClause =
+        participantAllDivulgedContractsPrunedUpToInclusive(connection) match {
+          case Some(pruneAfter) => cSQL"and event_offset > $pruneAfter"
+          case None => cSQL""
+        }
+
+      pruneWithLogging(queryDescription = "Immediate divulgence events pruning") {
+        SQL"""
+            -- Immediate divulgence pruning
+            delete from participant_events_create c
+            where event_offset <= $pruneUpToInclusive
+            -- Only prune create events which did not have a locally hosted party before their creation offset
+            and not exists (
+              select 1
+              from party_entries p
+              where p.typ = 'accept'
```

Comment:
was this the issue with the string comparison? if yes, would be nice to add a short comment that without this some fields got empty which lead to oracle issues etcetcetc

Reply:
If you are referring to the Oracle-specific blocker that I've mentioned offline, it was the comparison `participant_all_divulged_contracts_pruned_up_to_inclusive` (which was nullable/empty-stringed) to the `event_offset`. And in Oracle `''` does not compare lexicogrphically as expected.

Reply:
Which makes me wonder: this is a very common comparison that we're doing, howcome it did not pop out in other places 🤔 ?